SwiftIOSExperiments/Modules/Profile/Presenter/ProfilePresenter.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  ProfilePresenter.swift
+//  SwiftIOSExperiments
+//
+//  Created by James Santos on 16/2/17.
+//  Copyright © 2017 James Santos. All rights reserved.
+//
+
+import UIKit
+import Auth0
+
+class ProfilePresenter: ProfilePresentation {
+	weak var view: ProfileView?
+	var interactor: ProfilesUseCase!
+	var router: ProfilesWireframe!
+
+	var profile: Profile? = nil {
+		didSet {
+			if profile != nil {
+				view?.showProfileData(profile!)
+			}
+		}
+	}
+
+	func viewDidLoad() {
+		interactor.fetchProfile(AuthService.token!)
+	}
+}
+
+extension ProfilePresenter: ProfilesInteractorOutput {
+
+	func profileFetched(_ profile: Profile) {
+		self.profile = profile
+	}
+
+	func profileFetchedFailed(_ error: Error) {
+		print("Error: \(error.localizedDescription)")
+	}
```

Comment:
for this minimialist implementation you can use http://nshipster.com/swift-default-protocol-implementations/